mesonbuild/build.py

```diff
@@ -857,6 +858,8 @@ def get_extra_args(self, language):
         return self.extra_args.get(language, [])
 
     def get_dependencies(self, exclude=None, internal=True):
+        if self.cached_dependencies is not None:
+            return self.cached_dependencies
```

Comment:
This is also technically wrong, because the list of dependencies will be different based on whether or not `exclude` is set and `internal` is set. Not sure why the tests don't fail.

This also conflicts with https://github.com/mesonbuild/meson/pull/3939.

Reply:
Ah, good point. 

I'll leave this for a while, and see if this area is still a hotspot in profiling after #3939 is merged.

Reply:
I added a commit for caching of dependencies in #3939, can you try that?